app/src/views/swap.tsx

```diff
@@ -0,0 +1,245 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+// import './App.css'
+
+import React from 'react'
+
+// Context
+import { useSwapContext } from '../context/swap.context'
+
+// Hooks
+import { useWalletState } from '../state/wallet'
+
+// Types
+import { BlockchainToken, QuoteOption } from '../constants/types'
+
+// Components
+import { StandardButton, FlipTokensButton } from '../components/buttons'
+import {
+  SwapContainer,
+  FromSection,
+  ToSection,
+  SelectTokenModal,
+  QuoteOptions
+} from '../components/swap'
+import { SwapSectionBox } from '../components/boxes'
+
+// Assets
+import AdvancedIcon from '../assets/advanced-icon.svg'
+
+// Styled Components
+import { Row, Text, IconButton } from '../components/shared.styles'
+
+export const Swap = () => {
+  // Context
+  const { getLocale, getSwapQuotes } = useSwapContext()
+
+  // State
+  const {
+    state: { tokenBalances, tokenList, selectedNetwork, isConnected }
+  } = useWalletState()
+
+  // ToDo: Setup useSwap hook where all this kind of state will be handled.
+  const price = 1519.28
+  const rate = 0.026
+
+  // State
+  const [fromToken, setFromToken] = React.useState<
+    BlockchainToken | undefined
+  >()
+  const [toToken, setToToken] = React.useState<BlockchainToken | undefined>()
+  const [fromAmount, setFromAmount] = React.useState<string>('')
+  const [toAmount, setToAmount] = React.useState<string>('')
+  const [isFetchingQuote, setIsFetchingQuote] = React.useState<
+    boolean | undefined
+  >(undefined)
+  const [selectingFromOrTo, setSelectingFromOrTo] = React.useState<
+    'from' | 'to' | undefined
+  >(undefined)
+  const [quoteOptions, setQuoteOptions] = React.useState<QuoteOption[]>([])
+  const [selectedQuoteOption, setSelectedQuoteOption] = React.useState<
+    QuoteOption | undefined
+  >()
+
+  // Update on render
+  if (fromToken === undefined && fromToken !== tokenList[0]) {
+    setFromToken(tokenList[0])
+  }
+
+  // Fetch Quote when needed
+  if (
+    toToken !== undefined &&
+    fromToken !== undefined &&
+    fromAmount !== '' &&
+    quoteOptions.length === 0 &&
+    selectedQuoteOption === undefined &&
+    isFetchingQuote === undefined
+  ) {
+    setIsFetchingQuote(true)
+    getSwapQuotes(
+      fromToken.contractAddress,
+      fromAmount,
+      toToken.contractAddress
+    )
+      .then((result) => {
+        setQuoteOptions(result)
+        setSelectedQuoteOption(result[0])
+        setToAmount((Number(fromAmount) / rate).toFixed(6).toString())
+        setIsFetchingQuote(false)
+      })
+      .catch((error) => console.log(error))
+  }
```

Comment:
Data fetching should go in an action handler or useEffect with cleanup handlers
https://beta.reactjs.org/learn/you-might-not-need-an-effect#fetching-data

Reply:
Moved into a useEffect Hook!